packages/core/upload/admin/src/components/BulkMoveDialog/BulkMoveDialog.js

```diff
@@ -0,0 +1,135 @@
+import { Formik } from 'formik';
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { Button } from '@strapi/design-system/Button';
+import { Grid, GridItem } from '@strapi/design-system/Grid';
+import {
+  ModalLayout,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+} from '@strapi/design-system/ModalLayout';
+import { FieldLabel } from '@strapi/design-system/Field';
+import { Stack } from '@strapi/design-system/Stack';
+import { Typography } from '@strapi/design-system/Typography';
+import { VisuallyHidden } from '@strapi/design-system/VisuallyHidden';
+import { Form } from '@strapi/helper-plugin';
+
+import { getTrad } from '../../utils';
+import SelectTree from '../SelectTree';
+import { useFolderStructure } from '../../hooks/useFolderStructure';
+
+export const BulkMoveDialog = ({ onClose, errors: initialErrors }) => {
+  const submitButtonRef = useRef(null);
+  const { formatMessage } = useIntl();
+  const { data: folderStructure, isLoading } = useFolderStructure();
+  const initialFormData = {
+    destination: folderStructure[0],
+  };
+
+  const handleSubmit = values => {
+    onClose({ moved: true, destinationFolderId: values.destination.value });
+  };
+
+  const handleClose = () => {
+    onClose();
+  };
+
+  return (
+    <ModalLayout onClose={handleClose} labelledBy="title">
+      <ModalHeader>
+        <Typography fontWeight="bold" textColor="neutral800" as="h2" id="title">
+          {formatMessage({
+            id: getTrad('modal.folder.move.title'),
+            defaultMessage: 'Move elements to',
+          })}
+        </Typography>
+      </ModalHeader>
+
+      <ModalBody>
+        <Formik
+          validateOnChange={false}
+          onSubmit={handleSubmit}
+          initialValues={initialFormData}
+          initialErrors={initialErrors}
+        >
+          {({ values, errors, setFieldValue }) => (
+            <Form noValidate>
+              <Grid gap={4}>
+                <GridItem xs={12} col={12}>
+                  <Stack spacing={1}>
+                    <FieldLabel htmlFor="folder-destination">
+                      {formatMessage({
+                        id: getTrad('form.input.label.folder-location'),
```

Comment:
Yeah, tbh at the beginning I didn't add them to the translation files because everything worked without it and it feels a bit odd to add the default message again there. How do we handle this overall? Do we always put them into the files?

Do we have a way to collect them all at once? If so this might be worth doing on the feature branch at the end ...

Reply:
not sure I understood everything but yes indeed it works because it relates on the default message - the issue is if we forget to add them in the files is that it won't be possible to translate them - it will always rely on the default message which will stay in English.
we could do it at the end for this time if you want but overall I think we try to do it every time we make a change / on every PR to make sure we don't forget some! WDYT?

Reply:
While I agree this should be done per PR, I'd say I need to check the media library as a whole to see what I have missed and would like to open a separate PR for it, if that is ok for you?

Reply:
sounds good!